src/platform/posix/posix_clock.c

```diff
@@ -19,13 +19,45 @@
 
 #ifndef NNG_USE_GETTIMEOFDAY
 
-// Use POSIX realtime stuff
+#if __APPLE__
+#include <mach/mach.h>
+#include <mach/mach_time.h>
+#endif
+
+// Use POSIX realtime stuff                                                          
 nni_time
 nni_clock(void)
 {
 	struct timespec ts;
 	nni_time        msec;
 
+#if __APPLE__
+
+	int supported = 0;
+
+#if defined(MAC_OS_X_VERSION_10_12) && __has_builtin(__builtin_available)
```

Comment:
it also might make sense to completely remove the check when its not needed

```
#if (TARGET_OS_OSX && __MAC_OS_X_VERSION_MIN_ALLOWED < 101200)
// include support for clock_gettime fallback as we are targeting pre 10.12
#endif
```

this would remove the need to check for MAC_OS_X_VERSION_10_12, as were not trying to make it compile on older SDKs just be compatible with older OSs.

we also shouldn't need to check that __builtin_available is available, as were only apple which has that feature.

Reply:
Well, a part of the idea was to push this upstream at some point and the thinking is that they would like to see it compile on older SDKs.

Reply:
Also, I've seen the check for `__builtin_available` in many places, I'm guessing it's also to support older SDKs or some such thing, I did not get into it much, it's clear and cheap, so I just let it be.

Reply:
the code as it stands will not compile on < SDK 10.12 due to clock_gettime not even being defined.
I don't think we need to address that, as all were trying to address is the targeting of OS version < 10.12

so im propossing this

```
#if __APPLE__
#if (TARGET_OS_OSX && __MAC_OS_X_VERSION_MIN_ALLOWED < 101200)
    int has_clock_gettime = 0;
    if (__builtin_available(macOS 10.12, *)) {
         has_clock_gettime = 1;
    }
    if( ! has_clock_gettime ) {
        ....
    } else
#endif
#endif
```

Reply:
Their original code can build on older SDKs if you opt to use `gettimeofday()` instead of `clock_gettime()`, but we don't want to do that, cause it's prone to time-traveling problems (adjusting time).